scripts/partition_manager_report.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+import argparse
+import yaml
+import platform
+
+def print_region(region, size, reqs):
+    # Prepare colors (color code taken from size_report)
+    bcolors_ansi = {
+        "HEADER"    : '\033[95m',
+        "OKBLUE"    : '\033[94m',
+        "OKGREEN"   : '\033[92m',
+        "WARNING"   : '\033[93m',
+        "FAIL"      : '\033[91m',
+        "ENDC"      : '\033[0m',
+        "BOLD"      : '\033[1m',
+        "UNDERLINE" : '\033[4m'
+    }
+    if platform.system() == "Windows":
+        # Set all color codes to empty string on Windows
+        bcolors = dict.fromkeys(bcolors_ansi, '')
+    else:
+        bcolors = bcolors_ansi
+
+    # Print header
+    print(bcolors["OKBLUE"] + "%s (0x%x):" % (region, size) + bcolors["ENDC"])
+
+    # Sort partitions three times:
+    #  1. On whether they are a container (has a 'span'), containers first.
+    #  2. On size, descending.
+    #  3. On address, ascending.
+    sorted_reqs = sorted(sorted(sorted(reqs.keys(), key=lambda x: int('span' in reqs[x]), reverse=True), key=lambda x: reqs[x]['size'], reverse=True), key=lambda x: reqs[x]['address'])
+
+    # Create text lines
+    sizes = ["%s%s (0x%x)" % ((("| " + bcolors["WARNING"] if 'span' not in reqs[part] else "+---" + bcolors["OKBLUE"]) + "0x%x: " % reqs[part]['address']), part, reqs[part]['size']) + bcolors["ENDC"]  for part in sorted_reqs]
+    maxlen = max(map(len, sizes)) + 1
+
+    # Add top and bottom of frame. Add dummy color so alignment always works.
+    top_bottom = "+" + bcolors["OKBLUE"] + bcolors["ENDC"]
+    sizes = [top_bottom, *sizes, top_bottom]
+
+    # Print left-justified, framed lines
+    list(map(lambda s: print('%s' % s.ljust(maxlen, " ") + '|' if s[0] != '+' else s.ljust(maxlen, "-") + '+'), sizes))
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Parse given Partition Manager output YAML file and print a pretty report',
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument("-i", "--input", required=True, type=str,
+                        help="Path to the YAML file from Partition Manager")
+
+    args = parser.parse_args()
+
+    return args
+
+def main():
+    print("Partition Manager report")
+    args = parse_args()
+    with open(args.input, 'r') as f:
+        pm_config = yaml.safe_load(f)
+    min_address = min((part['address'] for part in pm_config.values() if 'address' in part))
+    max_address = max((part['address'] + part['size'] for part in pm_config.values() if 'address' in part))
+    print_region('FLASH', max_address - min_address, pm_config)
+
```

Comment:
2 lines before if check, same for main declaration

Reply:
will fix